io.openems.edge.evcs.goe.chargerhome/src/io/openems/edge/evcs/goe/chargerhome/GoeChargerHomeImpl.java

```diff
@@ -0,0 +1,340 @@
+			}
+			
+			break;
+		}
+	}
+	
+	private int ConvertGoeStatus(int status) {
+		switch(status) {
+		case 1: // ready for charging, car unplugged
+			return Status.NOT_READY_FOR_CHARGING.getValue();
+		case 2: // charging
+			return Status.CHARGING.getValue();
+		case 3: // waiting for car
+			return Status.READY_FOR_CHARGING.getValue(); 
+		case 4: // charging finished, car plugged
+			return Status.CHARGING_FINISHED.getValue();
+		default:
+			return Status.UNDEFINED.getValue();
+		}
+	}
+	
+	private int ConvertGoePhase(int phase) {
+		int Phasen = (byte)phase & 0b00111000;			
+		switch(Phasen) {
+		case 8: // 0b00001000: Phase 1 is active
+			return 1;
+		case 24: // 0b00011000: Phase 1+2 is active
+			return 2;
+		case 56: // 0b00111000: Phase1-3 are active
+			return 3;
+		default:
+			return 0;
+		}
+	}
+	
+	private int GetMaxHardwarePower(int cableCurrent) {
+		int MaxPower = 0;
+		int phases = 3;
+		if (cableCurrent < this.MaxCurrent && cableCurrent > 0) {
+			MaxPower = phases*cableCurrent*230/1000;
+		}
+		else {
+			MaxPower = phases*this.MaxCurrent*230/1000;
+		}
+		return MaxPower;
+	}
+	
+	private int GetMinHardwarePower(int phases, int cableCurrent) {
+		int MinPower = 0;
+		int MinPowerHw = phases*this.MinCurrent*230/1000;
+		if (cableCurrent < this.MinCurrent && cableCurrent > 0) {
+			MinPower = phases*cableCurrent*230/1000;
+		}
+		else {
+			MinPower = phases*this.MinCurrent*230/1000;
+		}
+		if (MinPower > MinPowerHw || MinPower < 0) {
+			return MinPowerHw;
+		}
+		else {
+			return MinPower;
+		}
+	}
+		
+	private int GetMaxPower(int phases) {
+		return phases*this.MaxCurrent*230/1000;
+	}
+	
+	/**
+	 * Sets the current from SET_CHARGE_POWER channel.
+	 * 
+	 * <p>
+	 * Allowed loading current are between MinCurrent and MaxCurrent. Invalid values are
+	 * discarded. The value is also depending on the DIP-switch settings and the
+	 * used cable of the charging station.
+	 */
+	private void setPower() {
+		
+		WriteChannel<Integer> energyLimitChannel = this.channel(ManagedEvcs.ChannelId.SET_ENERGY_LIMIT);
+		int energyLimit = energyLimitChannel.getNextValue().orElse(0);
+
+		// Check energy limit
+		if (energyLimit == 0 || energyLimit > this.getEnergySession().orElse(0)) {
+			WriteChannel<Integer> channel = this.channel(ManagedEvcs.ChannelId.SET_CHARGE_POWER_LIMIT);
+			Optional<Integer> valueOpt = channel.getNextWriteValueAndReset();
+			if (valueOpt.isPresent()) {
+	
+				Integer power = valueOpt.get();
+				Channel<Integer> minimumHardwarePowerChannel = this.channel(Evcs.ChannelId.MINIMUM_HARDWARE_POWER);
+				if (power < minimumHardwarePowerChannel.value()
+						.orElse(0)) { /* charging under MINIMUM_HARDWARE_POWER isn't possible */
+					power = 0;
+					this.goeapi.setActive(false);
+				}
+				else {
+					this.goeapi.setActive(true);
+				}
+				
+				Value<Integer> phases = this.getPhases();
+				Integer current = power * 1000 / phases.orElse(3) /* e.g. 3 phases */ / 230; /* voltage */
```

Comment:
The power we get should be already in W, so "*1000" is not needed.

Reply:
Sure, but the unit for the Charger-current is mA. So i've to multiply with 1000